web/pandas/pdeps/0005-no-default-index-mode.md

```diff
@@ -0,0 +1,377 @@
+- a ``MultiIndex`` could not be created with a ``NoRowIndex`` as one of its levels.
+
+Let's go into more detail for some of these. In the examples that follow, the ``NoRowIndex`` will be passed explicitly,
+but this is not how users would be expected to use it (see "Usage and Impact" section for details).
+
+### NoRowIndex.append
+
+If one has two ``DataFrame``s with ``NoRowIndex``, then one would expect that concatenating them would
+result in a ``DataFrame`` which still has ``NoRowIndex``. To do this, the following rule could be introduced:
+
+> If appending a ``NoRowIndex`` of length ``y`` to a ``NoRowIndex`` of length ``x``, the result will be a
+  ``NoRowIndex`` of length ``x + y``.
+
+Example:
+
+```python
+In [6]: df1 = pd.DataFrame({'a': [1,  2], 'b': [4, 5]}, index=NoRowIndex(2))
+
+In [7]: df2 = pd.DataFrame({'a': [4], 'b': [0]}, index=NoRowIndex(1))
+
+In [8]: df1
+Out[8]:
+ a  b
+ 1  4
+ 2  5
+
+In [9]: df2
+Out[9]:
+ a  b
+ 4  0
+
+In [10]: pd.concat([df1, df2])
+Out[10]:
+ a  b
+ 1  4
+ 2  5
+ 4  0
+
+In [11]: pd.concat([df1, df2]).index
+Out[11]: NoRowIndex(len=3)
+```
+
+Appending anything other than another ``NoRowIndex`` would raise.
+
+### Slicing a ``NoRowIndex``
+
+If one has a ``DataFrame`` with ``NoRowIndex``, then one would expect that a slice of it would still have
+a ``NoRowIndex``. This could be accomplished with:
+
+> If a slice of length ``x`` is taken from a ``NoRowIndex`` of length ``y``, then one gets a
+ ``NoRowIndex`` of length ``x``. Label-based slicing would not be allowed.
+
+Example:
+
+```python
+In [12]: df = pd.DataFrame({'a': [1,  2, 3], 'b': [4, 5, 6]}, index=NoRowIndex(3))
+
+In [13]: df.loc[df['a']>1, 'b']
+Out[13]:
+5
+6
+Name: b, dtype: int64
+
+In [14]: df.loc[df['a']>1, 'b'].index
+Out[14]: NoRowIndex(len=2)
+```
+
+Slicing by label, however, would be disallowed:
+```python
+In [15]: df.loc[0, 'b']
+---------------------------------------------------------------------------
+IndexError: Cannot use label-based indexing on NoRowIndex!
+```
+
+Note too that:
+- other uses of ``.loc``, such as boolean masks, would still be allowed (see F.A.Q);
+- ``.iloc`` and ``.iat`` would keep working as before;
+- ``.at`` would raise.
+
+### Aligning ``NoRowIndex``s
+
+To minimise surprises, the rule would be:
+
+> A ``NoRowIndex`` can only be aligned with another ``NoRowIndex`` of the same length.
+> Attempting to align it with anything else would raise.
+
+Example:
+```python
+In [1]: ser1 = pd.Series([1, 2, 3], index=NoRowIndex(3))
+
+In [2]: ser2 = pd.Series([4, 5, 6], index=NoRowIndex(3))
+
+In [3]: ser1 + ser2  # works!
+Out[3]:
+5
+7
+9
+dtype: int64
+
+In [4]: ser1 + ser2.iloc[1:]  # errors!
+---------------------------------------------------------------------------
+TypeError: Cannot join NoRowIndex of different lengths
+```
+
+### Columns cannot be NoRowIndex
+
+This proposal deals exclusively with letting users not have to think about
```

Comment:
Better to write "This proposal deals exclusively with allowing users to not have to think about"